luarules/gadgets/unit_minesweeper_detection.lua

```diff
@@ -42,82 +49,58 @@ for _, teamID in pairs(Spring.GetTeamList()) do
+function gadget:MetaUnitAdded(unitID, unitDefID, unitTeam)
 	if minesweeperRanges[unitDefID] then
 		minesweepers[unitTeam][unitID] = unitDefID
 	end
 end
 
-function gadget:UnitTaken(unitID, UnitDefID, oldTeam, newTeam)
-	if minesweeperRanges[unitDefID] then
-		minesweepers[oldTeam][unitID] = nil
-		minesweepers[newTeam][unitID] = unitDefID
-	end
-end
-
-function gadget:UnitDestroyed(unitID, unitDefID, unitTeam)
+function gadget:MetaUnitRemoved(unitID, unitDefID, unitTeam)
 	if minesweeperRanges[unitDefID] and minesweepers[unitTeam][unitID] then
 		minesweepers[unitTeam][unitID] = nil
 	end
 end
 
-local function processesTeamsSweepers(currentTeam)
+local function processTeamsSweepers(teamID)
 	local minesToReveal = {}
 
-	for sweeper, sweeperDef in pairs(minesweepers[currentTeam])  do
-		local x, y, z = Spring.GetUnitPosition(sweeper)
+	for sweeper, sweeperDef in pairs(minesweepers[teamID])  do
+		local x, y, z = spGetUnitPosition(sweeper)
 		if x and z then
-			local nearUnits = Spring.GetUnitsInCylinder(x, z, minesweeperRanges[sweeperDef])
+			local nearUnits = spGetUnitsInCylinder(x, z, minesweeperRanges[sweeperDef])
 			for _, nearUnit in ipairs(nearUnits) do
-				local nearUnitDefID = Spring.GetUnitDefID(nearUnit)
-				if currentTeam ~= Spring.GetUnitTeam(nearUnit) then
-					if mineIDs[nearUnitDefID] then
-						minesToReveal[nearUnit] = true
-					end
+				local nearUnitDefID = spGetUnitDefID(nearUnit)
+				if mineDefIDs[nearUnitDefID] and teamID ~= spGetUnitTeam(nearUnit) then
+					minesToReveal[nearUnit] = true
 				end
 			end
 		end
 	end
 
 	for mineToReveal, _ in pairs(minesToReveal) do
-		if revealedMines[currentTeam][mineToReveal] == nil then
-			revealedMines[currentTeam][mineToReveal] = true
+		if revealedMines[teamID][mineToReveal] == nil then
+			revealedMines[teamID][mineToReveal] = true
 			--show in vision and radar, and prevent engine from immediately resseting that state
-			Spring.Echo("revleaed", mineToReveal, "to", currentTeam)
-			Spring.SetUnitLosState(mineToReveal, teamIDs[currentTeam], 3)
-			Spring.SetUnitLosMask(mineToReveal, teamIDs[currentTeam], 3)
+			spSetUnitLosState(mineToReveal, teamIDs[teamID], 3)
+			spSetUnitLosMask(mineToReveal, teamIDs[teamID], 3)
 		end
 	end
 end
 
 local function processRevealedMines(teamID)
 	for revealedMine, _ in pairs(revealedMines[teamID]) do
-		Spring.SetUnitLosMask(revealedMine, teamID, 0)
+		spSetUnitLosMask(revealedMine, teamID, 0)
 		revealedMines[teamID][revealedMine] = nil
 	end
 end
 
--- if we have more teams than frames per second we use a slightly slower method
-local GAMESPEED = Game.gameSpeed
-if #teamIDs + 1 >= GAMESPEED then
-	function gadget:GameFrame(f)
-		local loopedFrame = f % GAMESPEED
-		for teamID in pairs(teamIDs) do
-			if teamID % GAMESPEED == loopedFrame then
-				processRevealedMines(teamID)
-				processesTeamsSweepers(teamID)
-			end
+local processingInterval = Game.gameSpeed
+function gadget:GameFrame(f)
```

Comment:
The convention, if it can even be said to exist, is to use `n` as the single letter variable name, or even better, `gameframe`

Reply:
where were all these complaints when hornet submitted the previous version

Reply:
wait, to you want me to replace the game speed variable with n, or the frame variable with n, or is it the one line bellow taht isn't in the highligh with n?

Reply:
`function gadget:GameFrame(n)`